core/src/main/java/org/apache/iceberg/CatalogProperties.java

```diff
@@ -30,6 +30,14 @@ private CatalogProperties() {
   public static final String FILE_IO_IMPL = "io-impl";
   public static final String WAREHOUSE_LOCATION = "warehouse";
 
+  public static final String TABLE_CACHE_ENABLED = "cache-enabled";
+  public static final boolean TABLE_CACHE_ENABLED_DEFAULT = true;
+  public static final String TABLE_CACHE_EXPIRATION_ENABLED = "cache.expiration-enabled";
```

Comment:
What about having a single property? We could use the expiration interval to disable the cache by setting it to 0.

Reply:
Sure that makes sense to me. Since it presently has to be set on the catalog configs, I wasn't that keen on having too many additional catalog configuration requirements.

Reply:
Users should probably not set this unless they want a specific value. If they don't know, they should omit it.

Reply:
I'm ok with that. To be clear, this would thus turn it on by default?

Reply:
Yes, I think caching should be on by default. That's the current default for `cache-enabled` anyway.

One thing I missed originally is that the `cache-enabled` property is already used in released versions. So we should continue to support disabling the cache using it. We should also disable the cache if the expiration is 0.

Reply:
I'm +1 for using a single property to control whether expiration is enabled and how long the expiration time is.